sigstore-java/src/main/java/dev/sigstore/trustroot/SigstoreTrustedRoot.java

```diff
@@ -48,4 +56,114 @@ static SigstoreTrustedRoot from(TrustedRoot proto) throws CertificateException {
         .cTLogs(ctlogs)
         .build();
   }
+
+  /**
+   * Find transparency log matching the logId and validity time.
+   *
+   * @param logId a log id
+   * @param time a point in time during the logs validity period
+   * @return the first log found matching logId and time, or empty if none match
+   */
+  public Optional<TransparencyLog> getTlog(byte[] logId, Instant time) {
+    return getTLog(getTLogs(), logId, time);
+  }
+
+  /**
+   * Find certificate transparency log matching the logId and validity time.
+   *
+   * @param logId a log id
+   * @param time a point in time during the logs validity period
+   * @return the first log found matching logId and time, or empty if none match
+   */
+  public Optional<TransparencyLog> getCTLog(byte[] logId, Instant time) {
+    return getTLog(getCTLogs(), logId, time);
+  }
+
+  private Optional<TransparencyLog> getTLog(
+      List<TransparencyLog> tlogs, byte[] logId, Instant time) {
+    return tlogs.stream()
+        .filter(tl -> Arrays.equals(tl.getLogId().getKeyId(), logId))
+        .filter(tl -> tl.getPublicKey().getValidFor().getStart().compareTo(time) <= 0)
+        .filter(
+            tl ->
+                tl.getPublicKey().getValidFor().getEnd().orElse(Instant.now()).compareTo(time) >= 0)
+        .findAny();
+  }
+
+  /**
+   * Get the one an only current TLog
+   *
+   * @return the current active TLog
+   * @throws IllegalStateException if trust root does not contain exactly one transparency log
+   */
+  public Optional<TransparencyLog> getCurrentTLog() {
```

Comment:
why is this optional then?

Reply:
Yeah it shouldn't be optional. I guess I never updated the signatures.